renaissance/scripts/perf/renaissance-run.sh

```diff
@@ -0,0 +1,301 @@
+# input:Return value of previous command
+# output:Prompts the error message if the return value is not zero 
+function err_exit() {
+	if [ $? != 0 ]; then
+		printf "$*"
+		echo
+		echo "The run failed. See setup.log for more details"
+		${K_EXEC} get pods -n ${NAMESPACE} >> ${LOGFILE}
+                ${K_EXEC} get events -n ${NAMESPACE} >> ${LOGFILE}
+                ${K_EXEC} logs pod/`${K_EXEC} get pods | grep "${APP_NAME}" | cut -d " " -f1` -n ${NAMESPACE} >> ${LOGFILE}
+		echo "1 , 99999 " >> ${RESULTS_DIR_ROOT}/Metrics-prom.log
+		cat ${RESULTS_DIR_ROOT}/Metrics-prom.log 
+		## Cleanup all the deployments
+		${SCRIPT_REPO}/renaissance-cleanup.sh -c ${CLUSTER_TYPE} -n ${NAMESPACE} >> ${LOGFILE}
+		exit 1
+	fi
+}
+# Run the benchmark as
+# SCRIPT BENCHMARK_SERVER_NAME NAMESPACE RESULTS_DIR_PATH WARMUPS MEASURES TOTAL_INST TOTAL_ITR RE_DEPLOY DURATION BENCHMARK
+# Ex of ARGS : --clustertype=openshift -s example.in.com -e /tfb/results -w 5 -m 3 -i 1 --iter=1 -r -d 60 -b page-rank
+# Describes the usage of the script
+function usage() {
+	echo
+	echo "Usage: $0 --clustertype=CLUSTER_TYPE -s BENCHMARK_SERVER -e RESULTS_DIR_PATH [-w WARMUPS] [-m MEASURES] [-i TOTAL_INST] [--iter=TOTAL_ITR] [-r= set redeploy to true] [-n NAMESPACE] [-g RENAISSANCE_IMAGE] [--cpureq=CPU_REQ] [--memreq=MEM_REQ] [--cpulim=CPU_LIM] [--memlim=MEM_LIM] [-b BENCHMARKS] [-R REPETITIONS] [-d DURATION] "
+	exit 1
+}
+
+# Iterate through the commandline options
+while getopts s:e:w:m:i:rg:n:t:R:d:b:-: gopts
+do
+	case ${gopts} in
+	-)
+		case "${OPTARG}" in
+			clustertype=*)
+				CLUSTER_TYPE=${OPTARG#*=}
+				;;
+			iter=*)
+				TOTAL_ITR=${OPTARG#*=}
+				;;			
+			cpureq=*)
+				CPU_REQ=${OPTARG#*=}
+				;;
+			memreq=*)
+				MEM_REQ=${OPTARG#*=}
+				;;
+			cpulim=*)
+				CPU_LIM=${OPTARG#*=}
+				;;
+			memlim=*)
+				MEM_LIM=${OPTARG#*=}
+				;;
+			envoptions=*)
+				ENV_OPTIONS=${OPTARG#*=}
+				;;
+			usertunables=*)
+                                OPTIONS_VAR=${OPTARG#*=}
+                                ;;
+			*)
+		esac
+		;;
+	s)
+		BENCHMARK_SERVER="${OPTARG}"
+		;;
+	e)
+		RESULTS_DIR_PATH="${OPTARG}"	
+		;;
+	w)
+		WARMUPS="${OPTARG}"		
+		;;
+	m)
+		MEASURES="${OPTARG}"		
+		;;
+	i)
+		TOTAL_INST="${OPTARG}"
+		;;
+	r)
+		RE_DEPLOY="true"
+		;;
+	g)
+		RENAISSANCE_IMAGE="${OPTARG}"		
+		;;
+	n)
+		NAMESPACE="${OPTARG}"		
+		;;
+	b)
+		BENCHMARKS="${OPTARG}"
+		;;
+	R)
+		REPETITIONS="${OPTARG}"
+		;;
+	d)
+		DURATION="${OPTARG}"
+		;;
+	esac
+done
+if [[ -z "${CLUSTER_TYPE}" || -z "${BENCHMARK_SERVER}" || -z "${RESULTS_DIR_PATH}" ]]; then
+	echo "Do set the variables - CLUSTER_TYPE, BENCHMARK_SERVER and RESULTS_DIR_PATH "
+	usage
+fi
+
+if [ -z "${WARMUPS}" ]; then
+	WARMUPS=5
+fi
+
+if [ -z "${MEASURES}" ]; then
+	MEASURES=3
+fi
+
+if [ -z "${TOTAL_INST}" ]; then
+	TOTAL_INST=1
+fi
+
+if [ -z "${TOTAL_ITR}" ]; then
+	TOTAL_ITR=1
+fi
+
+if [ -z "${RE_DEPLOY}" ]; then
+	RE_DEPLOY=false
+fi
+
+if [ -z "${BENCHMARK_IMAGE}" ]; then
```

Comment:
This should be RENAISSANCE_IMAGE

Reply:
This was made for a general benchmark of user's choice so kept the name as BENCHMARK_IMAGE